src/spauna-cli.ts

```diff
@@ -1,17 +1,23 @@
 import yargs, { Arguments } from "yargs";
 import { hideBin } from "yargs/helpers";
+import { runCommand } from "./spauna.js";
 
 yargs(hideBin(process.argv))
   .command(
     "serve",
     "start the server",
     (yargs) => {
-      return yargs.option("port", {
-        type: "number",
-      });
+      return yargs
+        .option("port", {
+          type: "number",
+        })
+        .option("config", {
+          type: "string",
+        })
+        .demandOption("config");
     },
-    (argv: Arguments) => {
-      console.log("time to serve", argv);
+    async (argv: Arguments) => {
+      await runCommand("serve", argv.config as string, argv);
```

Comment:
In this case, the types of `yargs.command()` are such that if you don't add a type to `argv` here the appropriate type is inferred.

```suggestion
    async (argv) => {
      await runCommand("serve", argv.config, argv);
```

Reply:
VS Code disagrees
<img width="748" alt="image" src="https://github.com/user-attachments/assets/088c0d86-4488-4a38-830e-b1a7b061c1a3">

Reply:
Odd... it worked for me:

![](https://github.com/user-attachments/assets/e1ccf792-2307-4a85-aad0-b016eee822e5)